### PR TITLE
Update docstring and readme

### PR DESCRIPTION
**Pull request by mssfang**
- added text input size to sync and async operation javadoc
- update readme

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ Text Analytics supports both [multi-service and single-service access][service_a
 resource if you plan to access multiple cognitive services under a single endpoint/key. For Text Analytics access only,
 create a Text Analytics resource.
 
-You can create either resource using the [Azure Portal][create_new_resource] or [Azure CLI][azure_cli].
+You can create either resource using the **Option 1:** [Azure Portal][create_new_resource] or **Option 2:** [Azure CLI][azure_cli].
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/README.md]: Might be nice to format the options on different lines. Here's how it looks now: https://github.com/Azure/azure-sdk-for-java/blob/564cc2cf9cf11318078014aa3d03314039730bcb/sdk/textanalytics/azure-ai-textanalytics/README.md#create-a-text-analytics-resource

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 2]
```diff
@@ -56,73 +56,30 @@ az cognitiveservices account create \
 ```
 ### Authenticate the client
 In order to interact with the Text Analytics service, you'll need to create an instance of the 
-`TextAnalyticsClient` class. You would need an **endpoint** and **API key** to instantiate a client object.
+`TextAnalyticsClient` class. You would need an **endpoint** and either an **API key** or **AAD TokenCredential** to instantiate a client object.
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/README.md]: "You will need..."

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 3]
```diff
@@ -134,17 +91,54 @@ TextAnalyticsClient textAnalyticsClient = new TextAnalyticsClientBuilder()
...
+See [service limitations][service_input_limitation] for the input, including document length limits, maximum batch size,
+and supported text encoding.
 
 ### Operation Result
 An operation result, such as `AnalyzeSentimentResult`, is the result of a Text Analytics operation, containing a 
 prediction or predictions about a single text input. An operation's result type also may optionally include information
-about the input document and how it was processed.
+about the input document and how it was processed. An operation result contains a `isError` property that allows to
+identify if an operation executed was successful or unsuccessful for the given text input. When the operation results
+an error. You can simply call `getError()` to get `TextAnalyticsError` which contains the reason why it is unsuccessful.
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/README.md]: "When the operation results an error." I think you meant to use a comma here and include the next sentence, but not sure.

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 4]
```diff
@@ -56,73 +56,30 @@ az cognitiveservices account create \
 ```
 ### Authenticate the client
 In order to interact with the Text Analytics service, you'll need to create an instance of the 
-`TextAnalyticsClient` class. You would need an **endpoint** and **API key** to instantiate a client object.
+`TextAnalyticsClient` class. You would need an **endpoint** and either an **API key** or **AAD TokenCredential** to instantiate a client object.
 And they can be found in the [Azure Portal][azure_portal] under the "Quickstart" in your created
-Text Analytics resource. 
+Text Analytics resource. See the full details regarding [authentication][authentication] of cognitive services.
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/README.md]: "Cognitive Services"

File: sdk/textanalytics/azure-ai-textanalytics/README.md (hunk 3 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/README.md]: yup.

File: sdk/textanalytics/azure-ai-textanalytics/README.md [hunk 5]
```diff
@@ -134,17 +91,54 @@ TextAnalyticsClient textAnalyticsClient = new TextAnalyticsClientBuilder()
...
 ### Text Input
-A text input, sometimes called a `document`, is a single unit of input to be analyzed by the predictive models
+A **text input**, sometimes called a **document**, is a single unit of input to be analyzed by the predictive models
 in the Text Analytics service. Operations on Text Analytics client may take a single text input or a collection
-of inputs to be analyzed as a batch.
+of inputs to be analyzed as a batch. 
+See [service limitations][service_input_limitation] for the input, including document length limits, maximum batch size,
+and supported text encoding.
 
 ### Operation Result
```
Comment (maririos) [inline sdk/textanalytics/azure-ai-textanalytics/README.md]: Consider changing to `Return value`

File: sdk/textanalytics/azure-ai-textanalytics/README.md (hunk 1 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/README.md]: updated

File: sdk/textanalytics/azure-ai-textanalytics/README.md (hunk 5 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/README.md]: so i guess "Operation Result Collection" will need to change to "Result Value Collection"?

**Pull request merged**